client/src/controllers/PreviewController.ts

```diff
@@ -0,0 +1,772 @@
+
+  /** Device size `<input type="radio">` elements. */
+  declare readonly sizeTargets: HTMLInputElement[];
+  declare readonly hasNewTabTarget: boolean;
+  /** New tab button. */
+  declare readonly newTabTarget: HTMLAnchorElement;
+  declare readonly hasSpinnerTarget: boolean;
+  /** Loading spinner. */
+  declare readonly spinnerTarget: HTMLDivElement;
+  declare readonly hasModeTarget: boolean;
+  /** Preview mode `<select>` element. */
+  declare readonly modeTarget: HTMLSelectElement;
+  /** The main preview `<iframe>` that is currently displayed. */
+  declare readonly iframeTarget: HTMLIFrameElement;
+  /** All preview `<iframes>` that are currently in the DOM.
+   * This contains the currently displayed `<iframe>` and may also contain
+   * the new `<iframe>` that will replace the current one. */
+  declare readonly iframeTargets: HTMLIFrameElement[];
+
+  /** URL for updating the preview data. Also used for rendering the preview if `renderUrlValue` is unset. */
+  declare readonly urlValue: string;
+  /** Interval in milliseconds when the form is checked for changes.
+   * Also used as the debounce duration for the update request. */
+  declare readonly autoUpdateIntervalValue: number;
+  /** CSS property for setting the device width. */
+  declare readonly deviceWidthPropertyValue: string;
+  /** CSS property for the current width of the panel, to maintain the device scaling. */
+  declare readonly panelWidthPropertyValue: string;
+  /** Key for storing the last selected device size in localStorage. */
+  declare readonly deviceLocalStorageKeyValue: string;
+  /** URL for rendering the preview, defaults to `urlValue`.
+   * Useful for headless setups where the front-end may be hosted at a different URL. */
+  declare renderUrlValue: string;
+
+  declare readonly hasWProgressOutlet: boolean;
+  /** ProgressController for the refresh button that may be displayed when auto-update is turned off. */
+  declare readonly wProgressOutlet: ProgressController;
+
+  // Instance variables with initial values set in connect()
+
+  /** Main editor form. */
+  declare editForm: HTMLFormElement;
+  /** Side panel element of the preview panel, i.e. the element with the
+   * `data-side-panel` attribute. Useful for listening to show/hide events.
+   * Normally, this is the parent element of the controller element.
+   */
+  declare sidePanelContainer: HTMLDivElement;
+  /** Side panel for content checks. */
+  declare checksSidePanel: HTMLDivElement | null;
+  /** ResizeObserver to observe when the panel is resized
+   * so we can maintain the device size scaling. */
+  declare resizeObserver: ResizeObserver;
+
+  // Instance variables with initial values set here
+
+  /** Timeout before displaying the loading spinner. */
+  spinnerTimeout: ReturnType<typeof setTimeout> | null = null;
+  /** Interval for the auto-update. */
+  updateInterval: ReturnType<typeof setOptionalInterval> = null;
+  /** Whether the preview is ready for further updates.
```

Comment:
This is a bit long but I'd like to make sure the edge cases that makes the "DELETE" request necessary are documented in the code.